modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/LogicalRelImplementor.java

```diff
@@ -880,6 +882,22 @@ else if (rel instanceof Intersect)
         return node;
     }
 
+    /** {@inheritDoc} */
+    @Override public Node<Row> visit(IgniteUncollect rel) {
+        UncollectNode<Row> node = new UncollectNode<>(
+            ctx,
+            rel.getInput().getRowType(),
+            rel.getRowType(),
+            rel.withOrdinality
+        );
+
+        Node<Row> input = visit(rel.getInput());
```

Comment:
Minor: can be `node.register(visit(rel.getInput()));`

Reply:
Fixed